packages/portfolio-contract/src/portfolio.exo.ts

```diff
@@ -160,74 +163,52 @@ export const preparePortfolioKit = (
-            if (Aave && targetChainId === Aave.chain) {
-              Aave.remoteAddressVK.resolver.resolve(address);
-            } else if (Compound && targetChainId === Compound.chain) {
-              Compound.remoteAddressVK.resolver.resolve(address);
-            } else {
-              throw makeError(
-                `no matching chainId ${targetChainId}: ${q({ Aave: Aave?.chain, Compound: Compound?.chain })}`,
-              );
-            }
+            const { Gmp } = this.state;
+            Gmp.remoteAddressVK.resolver.resolve(address);
+            trace(`remoteAddress ${address}`);
           }
 
           trace('receiveUpcall completed');
         },
       },
       keeper: {
+        getGMPAddress() {
+          const { Gmp } = this.state;
+          return Gmp.remoteAddressVK.vow;
+        },
         getLCA() {
-          const { localAccount } = this.state;
-          return localAccount;
+          const { Gmp } = this.state;
+          return Gmp.localAccount;
         },
         getPositions(): YieldProtocol[] {
           const { state } = this;
           return harden(
             (keys(YieldProtocol) as YieldProtocol[]).filter(k => !!state[k]),
           );
         },
+        getUSDNICA() {
+          return this.state.USDN;
+        },
         initAave(chain: CaipChainId) {
           this.state.Aave = harden({
             chain,
-            remoteAddressVK: vowTools.makeVowKit(),
           });
-          trace('initAave', chain);
         },
-        initCompound() {
+        initCompound(chain: CaipChainId) {
           this.state.Compound = harden({
-            chain: EthL1,
-            remoteAddressVK: vowTools.makeVowKit(),
+            chain,
           });
-          trace('initCompound');
-        },
-        initUSDN(account: NobleAccount) {
-          this.state.USDN = account;
-          trace('initUSDN', account);
-        },
-        getUSDNICA() {
-          const { USDN } = this.state;
-          if (!USDN) throw Error('USDN not set');
-          return USDN;
-        },
-        getAaveAddress() {
-          const { Aave } = this.state;
-          if (!Aave) throw Error('Aave not set');
-          return Aave.remoteAddressVK.vow;
         },
-        getCompoundAddress() {
-          const { Compound } = this.state;
-          if (!Compound) throw Error('Compound not set');
-          return Compound.remoteAddressVK.vow;
+        /** KLUDGE around lack of synchronization signals for now. TODO: rethink design. */
+        wait(val: bigint) {
+          return vowTools.watch(E(timer).delay(val));
         },
       },
       rebalanceHandler: {
-        async handle(seat: ZCFSeat, offerArgs: CopyRecord) {
+        async handle(seat: ZCFSeat, offerArgs: unknown) {
           const { keeper } = this.facets;
+          mustMatch(offerArgs, OfferArgsShapeFor.rebalance);
+          // @ts-expect-error: offerArgs is validated just above and safe to use as OfferArgsFor['rebalance']
```

Comment:
the `mustMatch` from `@agoric/internal` works with `TypedPattern<T>` and makes this `@ts-expect-error` unnecessary.